packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -53,16 +82,24 @@ async function loadChainId(baseUrl: string): Promise<ChainId> {
 }
 
 export class EthereumConnection implements BcpConnection {
-  public static async establish(baseUrl: string): Promise<EthereumConnection> {
+  public static async establish(baseUrl: string, wsUrl: string | undefined): Promise<EthereumConnection> {
     const chainId = await loadChainId(baseUrl);
-    return new EthereumConnection(baseUrl, chainId);
+    let socketStream;
+    if (wsUrl) {
+      socketStream = new StreamingSocket(wsUrl);
+      socketStream.connect();
+      await socketStream.connected;
+    }
+    return new EthereumConnection(baseUrl, chainId, socketStream);
```

Comment:
All this connection code cam move into constructor. We only need to load the ChainID here because this needs to be available in as a readonly member.